src/explore-education-statistics-admin/src/pages/bau/BoundaryDataUploadPage.tsx

```diff
@@ -0,0 +1,157 @@
+import boundaryDataService from '@admin/services/boundaryDataService';
+import React, { useCallback } from 'react';
+import Page from '@admin/components/Page';
+import FormProvider from '@common/components/form/FormProvider';
+import FormFieldset from '@common/components/form/FormFieldset';
+import FormGroup from '@common/components/form/FormGroup';
+import FormFieldDateInput from '@common/components/form/FormFieldDateInput';
+import Button from '@common/components/Button';
+import FormFieldFileInput from '@common/components/form/FormFieldFileInput';
+import {
+  Form,
+  FormFieldSelect,
+  FormFieldTextInput,
+} from '@common/components/form';
+import { useHistory } from 'react-router-dom';
+
+export default function BoundaryDataUploadPage() {
+  interface FormValues {
+    boundaryLevel: string;
+    boundaryLevelLabel: string;
+    boundaryLevelPublishedDate: Date;
+    boundaryDataFile: File;
+  }
+
+  const history = useHistory();
+  const boundaryDataPath = '/administration/boundary-data';
+
+  const handleSubmit = useCallback(
+    async (values: FormValues) => {
+      const data = {
+        level: values.boundaryLevel,
+        label: values.boundaryLevelLabel,
+        file: values.boundaryDataFile,
+        published: values.boundaryLevelPublishedDate,
+      };
+
+      await boundaryDataService.uploadBoundaryFile(data);
+      history.push(boundaryDataPath);
+    },
+    [boundaryDataPath, history],
+  );
+
+  return (
+    <Page
+      title="Boundary data"
+      caption="Upload boundary data"
+      wide
+      breadcrumbs={[
+        { name: 'Platform administration', link: '/administration' },
+        { name: 'Boundary data', link: '/administration/boundary-data' },
+        { name: 'Upload boundary data' },
+      ]}
+    >
+      <FormProvider enableReinitialize>
+        {({ formState }) => {
+          return (
+            <Form id="addBoundaryDataForm" onSubmit={handleSubmit}>
+              <FormFieldset
+                id="boundaryData"
+                legend="Upload new boundary data"
+                legendSize="l"
+                hint="This file will be processed and added to the list of available boundary files on publications."
+              >
+                <FormGroup>
+                  <FormFieldSelect<FormValues>
+                    id="boundaryLevel"
+                    name="boundaryLevel"
+                    label="Select boundary level"
+                    className="govuk-!-width-one-third"
+                    options={[
+                      { value: '', label: '' },
+                      {
```

Comment:
I'd also be inclined to use `locationLevelsMap` here.

Reply:
See comment on `BoundaryDataPage`.